cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1230,11 +1230,24 @@ protected CompositeFuture adminClientSecrets() {
         }
 
         Future<ReconciliationState> zkJmxSecret() {
+            Map<String, String> jmxSecretLabels = emptyMap();
+            Map<String, String> jmxSecretAnnotations = emptyMap();
+
+            if (kafkaAssembly.getSpec().getZookeeper() != null
+                    && kafkaAssembly.getSpec().getZookeeper().getTemplate() != null
+                    && kafkaAssembly.getSpec().getZookeeper().getTemplate().getJmxSecret() != null
+                    && kafkaAssembly.getSpec().getZookeeper().getTemplate().getJmxSecret().getMetadata() != null) {
+                jmxSecretAnnotations = kafkaAssembly.getSpec().getZookeeper().getTemplate().getJmxSecret().getMetadata().getAnnotations();
+                jmxSecretLabels = kafkaAssembly.getSpec().getZookeeper().getTemplate().getJmxSecret().getMetadata().getLabels();
+            }
             if (zkCluster.isJmxAuthenticated()) {
                 Future<Secret> secretFuture = secretOperations.getAsync(namespace, ZookeeperCluster.jmxSecretName(name));
+                Map<String, String> finalJmxSecretAnnotations = jmxSecretAnnotations;
+                Map<String, String> finalJmxSecretLabels = jmxSecretLabels;
                 return secretFuture.compose(secret -> {
                     if (secret == null) {
-                        return withVoid(secretOperations.reconcile(reconciliation, namespace, ZookeeperCluster.jmxSecretName(name), zkCluster.generateJmxSecret()));
+                        return withVoid(secretOperations.reconcile(reconciliation, namespace, ZookeeperCluster.jmxSecretName(name),
+                                zkCluster.generateJmxSecret(finalJmxSecretAnnotations, finalJmxSecretLabels)));
```

Comment:
I don't think this should be done in the `KafkaAssemblyOperator` class. This should be all encapsulated by the model classes (`KafkaCluster`, ZooKeeperCluster`, `AbstractModel` etc.). Same for the Kafka secret below. These classes already handle the template properties for many other resources, so you should be able to find there how it is done and used.